### DISCUSSION TITLE
Why are「,」on ],\ in Mozc Kana keyboard?

### DISCUSSION THREAD

**Discussion by HubKing** [fcitx/fcitx5#796] [Q&A]
I have not used Windows recently, but if I remember it correctly, the Japanese 「、」were shift + [,] on Windows. And if I remember it correctly, it was also like that on Fcitx5 Mozc before. But today, I have discovered that shift + [ enters ゛on Fcitx5. That looks the same thing that is typed when I press just [ in Kana mode. Is this a bug?

I just searched Google Images to see physical Japanese keyboards, and they all seem to have 「 on the same key as [, so I think it is weird that I need to type shift + ] to enter 「.

![image](https://github.com/fcitx/fcitx5/assets/10900489/10c79246-3da5-4472-8259-f03804b99d50) ![image](https://github.com/fcitx/fcitx5/assets/10900489/ab5d3138-09b6-4123-8f58-d423ed71d4d1) ![image](https://github.com/fcitx/fcitx5/assets/10900489/98a82426-f973-44e4-86a9-d3954555b8a2)

**Comment by HubKing** [comment]
PS: My physical keyboard layout is US English.

**Comment by lunarythia** [comment]
@HubKing Mozc is trying to treat your American keyboard like the Japanese physical keyboard layout (because direct kana input is made for Japanese computers), but the two layouts are different. They have only two keys to the right of `P`, but three to the right of `L` and `0`, and four to the right of `M`. (I’m not quite sure how you are entering `¥` and `ro`, actually)

On US keyboards, the key to the right of `P` is `[ {` like you said, but if you look at the Japanese keyboard, to the right of `P` is `゛`. If you look the next key over, for US it is `] }` but for the Japanese keyboard it is `°「`.

Slight tangent (still relevant), some European keyboards have, 2 keys to the right of `P`, 3 keys to the right of `L`, and 1 key between Left Shift and `Z`. But the reason I mention this is because the backslash key on our keyboards becomes that extra key near `L` on their keyboards.

Same thing with the japanese Keyboard. Even though the `mu 」` key doesn’t seem to exist on an US keyboard, it actually corresponds to our backslash key.
